src/classes/storage.ts

```diff
@@ -84,4 +100,155 @@ export class Storage {
+   * @param {AutoCompleteFilters[]|undefined} filters - The filters you want to be applied on the search.
+   * @returns {AutoComplete[]}
+   */
+  public filterGroundPlaces(filters?: AutoCompleteFilters[]): AutoComplete[] {
+    const groundPlacesAutocomplete: AutoComplete[] = this.parseGroundPlaces();
+
+    // Returns earlier the Ground places if there is no filters to use.
+    if (!filters || !filters.length) return groundPlacesAutocomplete;
+
+    const filteredByGroup: AutoComplete[] = this.filterGroundPlacesByGroup(
+      groundPlacesAutocomplete,
+      filters.includes(AutoCompleteFilters.STOP_GROUP),
+    );
+
+    const filteredByCluster: AutoComplete[] = this.filterGroundPlacesByCluster(
+      groundPlacesAutocomplete,
+      filters.includes(AutoCompleteFilters.STOP_CLUSTER),
+    );
+
+    const filteredByType: AutoComplete[] = [...filteredByGroup, ...filteredByCluster];
+
+    const filteredBySegmentProviderStop: AutoComplete[] = this.filterGroundPlacesBySegmentProvider(
+      // Filtering by segment provider must be made on the whole results previously filtered.
+      // We checks here if this filter must be applied on results already filtered or not.
+      filteredByType.length ? filteredByType : groundPlacesAutocomplete,
+      filters.includes(AutoCompleteFilters.SEGMENT_PROVIDER_STOP),
+    );
+
+    const filteredByService: AutoComplete[] = this.filterGroundPlacesByServiced(
+      filteredBySegmentProviderStop,
+      filters.includes(AutoCompleteFilters.SERVICED),
+    );
+
+    return filteredByService;
+  }
+
+  /**
+   * @description Search through ground places with specific words.
+   * @param {AutoComplete[]} groundPlaces - The places on which the search is used.
+   * @param {string} search - The search to find the requested places.
+   * @returns {AutoComplete[]}
+   */
+  public searchGroundPlaces(groundPlaces: AutoComplete[], search: string): AutoComplete[] {
+    return groundPlaces.filter((place) => {
+      // Method toLowerCase() is used because the includes() method is case sensitive
+      const currentSearch = search.toLowerCase();
+
+      /* Search by unique name. This search only concern StopCluster. */
+      if (place.type === GroundPlaceType.CLUSTER && place.unique_name.toLowerCase().includes(currentSearch)) {
+        return place;
+      }
+
+      /* Search by Gpuid and by name */
+      if (place.gpuid.toLowerCase().includes(currentSearch) || place.name.toLowerCase().includes(currentSearch)) {
+        return place;
+      }
+    });
+  }
+
+  // -----------------------------------------------------------------------------
+  // ----------------------------- PRIVATE METHODS -------------------------------
+  // -----------------------------------------------------------------------------
+
+  /**
+   * @description Converts the Ground places to an array list manipulable.
+   * @returns {AutoComplete[]}
+   */
+  private parseGroundPlaces(): AutoComplete[] {
```

Comment:
Not sure to understand that

Reply:
Yes it goes back to my comment above here https://github.com/tictactrip/ground-places-sdk/pull/3#discussion_r529121430.

To use research methods effectively, I need to use an array of objects instead of an object

Reply:
this a really nice idea of optimization if you maintain this structure to search through it. And only refresh it every changes.

But if you build it everytime this is worse.

My advice ? start simple. optimize later